File: app/src/main/java/pl/karol202/paintplus/tool/rubber/RubberProperties.java
/*
 * Copyright 2017 karol-202
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package pl.karol202.paintplus.tool.rubber;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.SeekBar;
import android.widget.TextView;
import pl.karol202.paintplus.R;
import pl.karol202.paintplus.tool.ToolProperties;
import pl.karol202.paintplus.util.SeekBarTouchListener;

public class RubberProperties extends ToolProperties implements SeekBar.OnSeekBarChangeListener, CompoundButton.OnCheckedChangeListener
{
	private ToolRubber rubber;
	
	private View view;
	private SeekBar seekRubberSize;
	private TextView textRubberSize;
	private CheckBox checkSmooth;
	
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
	{
		super.onCreateView(inflater, container, savedInstanceState);
		view = inflater.inflate(R.layout.properties_rubber, container, false);
		rubber = (ToolRubber) tool;
		
		seekRubberSize = view.findViewById(R.id.seekBar_rubber_size);
		seekRubberSize.setProgress((int) rubber.getSize() - 1);
		seekRubberSize.setOnSeekBarChangeListener(this);
		seekRubberSize.setOnTouchListener(new SeekBarTouchListener());
		
		textRubberSize = view.findViewById(R.id.rubber_size);
		textRubberSize.setText(String.valueOf(seekRubberSize.getProgress() + 1));
		
		checkSmooth = view.findViewById(R.id.check_rubber_smooth);
		checkSmooth.setChecked(rubber.isSmooth());
		checkSmooth.setOnCheckedChangeListener(this);
		return view;
	}
	
	@Override
	public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser)
	{
		setRubberSize(progress);
	}
	
	private void setRubberSize(int size)
	{
		rubber.setSize(size + 1);
		textRubberSize.setText(String.valueOf(size + 1));
	}
	
	@Override
	public void onStartTrackingTouch(SeekBar seekBar) { }
	
	@Override
	public void onStopTrackingTouch(SeekBar seekBar) { }
	
	@Override
	public void onCheckedChanged(CompoundButton buttonView, boolean isChecked)
	{
		rubber.setSmooth(isChecked);
	}
}

File: app/src/main/java/pl/karol202/paintplus/tool/rubber/RubberProperties.kt
/*
 * Copyright 2017 karol-202
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package pl.karol202.paintplus.tool.rubber

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import org.koin.android.ext.android.inject
import pl.karol202.paintplus.R
import pl.karol202.paintplus.databinding.PropertiesRubberBinding
import pl.karol202.paintplus.util.SeekBarTouchListener
import pl.karol202.paintplus.util.setOnValueChangeListener
import pl.karol202.paintplus.util.viewBinding

class RubberProperties : Fragment(R.layout.properties_rubber)
{
	private val toolRubber by inject<ToolRubber>()

	private val views by viewBinding(PropertiesRubberBinding::bind)

	@SuppressLint("ClickableViewAccessibility")
	override fun onViewCreated(view: View, savedInstanceState: Bundle?)
	{
		views.seekBarRubberSize.progress = toolRubber.size.toInt() - 1
		views.seekBarRubberSize.setOnValueChangeListener { setRubberSize(it) }
		views.seekBarRubberSize.setOnTouchListener(SeekBarTouchListener())

		views.textRubberSize.text = (views.seekBarRubberSize.progress + 1).toString()

		views.checkRubberSmooth.isChecked = toolRubber.smoothEdge
		views.checkRubberSmooth.setOnCheckedChangeListener { _, checked -> toolRubber.smoothEdge = checked }
	}

	private fun setRubberSize(size: Int)
	{
		toolRubber.size = (size + 1).toFloat()
		views.textRubberSize.text = (size + 1).toString()
	}
}
